src/core/DataFile.cpp

```diff
@@ -326,6 +336,139 @@ bool DataFile::writeFile( const QString& filename )
 
 
 
+// TODO: Remove qWarnings
+bool DataFile::writeBundle(const QString& filename)
+{
+	qWarning() << "Write Bundle DEBUG:";
+
+	const QString fullName = nameWithExtension(filename);
+	const QString fullNameTemp = fullName + ".new";
+	const QString resourcesDirName = QFileInfo(fullName).path() + "/resources";
+
+	qWarning() << "Destination File: " << fullName;
+	qWarning() << "Resources Folder: " << resourcesDirName;
+
+	QFile outfile(fullNameTemp);
+
+	if (!outfile.open(QIODevice::WriteOnly | QIODevice::Truncate))
+	{
+		if (gui)
+		{
+			QMessageBox::critical(NULL,
+				SongEditor::tr("Could not write file"),
+				SongEditor::tr("Could not open %1 for writing. You probably are not permitted to "
+								"write to this file. Please make sure you have write-access to "
+								"the file and try again.").arg(fullName));
+		}
+		qWarning() << "ERROR: Failed to create bundle file!";
+		return false;
+	}
+
+	// Creates resources folder
+	if (QDir(resourcesDirName).exists())
+	{
+		qWarning() << "Warning: Resources dir already exists. Proceeding anyways.";
+	}
+	else
+	{
+		if(!QDir().mkdir(resourcesDirName))
+		{
+			qWarning() << "ERROR: Failed to create resources dir!";
+			return false;
+		}
+	}
+
+	// Files will be named by the counter number + file extension
+	// to avoid name conflicts
+	unsigned int resCounter = 1;
+
+	std::map<QString,std::vector<QString>>::const_iterator it = ELEMENTS_WITH_RESOURCES.begin();
```

Comment:
Maybe we should make a `typedef` for this `std::map<QString, std::vector<QString>>`.

Reply:
Oh, forgot to mention, I added a typedef for this. Called it `ResourcesMap`!